File: ifif/python/G5_17609.py
import sys

sys.stdin = open("input.txt", "r")

T = int(input())
for _ in range(T):
    S = input()
    start = 0
    end = len(S)-1
    cnt = 0
    while start<=end:
        if S[start] == S[end]:
            start += 1
            end -= 1
            continue
        cnt += 1
        if (S[start] == S[end-1]) and (S[start+1] == S[end]):
            temp_start = start
            temp_end = end-1
            while temp_start <= temp_end:
                if (S[temp_start] == S[temp_end]):
                    temp_start += 1
                    temp_end -= 1
                    continue
                start += 1
                break
            else:
                end -= 1
        elif (S[start] == S[end-1]):
            end -= 1
        elif (S[start+1] == S[end]):
            start += 1
        if cnt == 2:
            break
    print(cnt, S)

File: ifif/java/G5_17609.java
import java.io.*;
import java.util.Scanner;

public class G5_17609 {
    public static void main(String[] args) throws IOException {
        System.setIn(new FileInputStream("input.txt"));
        BufferedReader sc = new BufferedReader(new InputStreamReader(System.in));
        int T = Integer.parseInt(sc.readLine());
        for(int t=0; t<T; t++){
            String S = sc.readLine();
            System.out.println(solution(S));
        }
    }

    private static int solution(String S) {
        if (isPal(S)) {
            return 0;
        }
        int cnt = 0;
        int start = 0;
        int end = S.length()-1;
        while (start<=end) {
            if (S.charAt(start) == S.charAt(end)) {
                start++;
                end--;
                continue;
            }

            cnt++;
            if (cnt>=2) {
                return 2;
            }
            if (S.charAt(start+1) != S.charAt(end)) {
                end--;
            } else if (S.charAt(start) != S.charAt(end-1)) {
                start++;
            }
            else {
              int temp_start = start+1;
              int temp_end = end;
              start++;
              while (temp_start<=temp_end) {
                  if (S.charAt(temp_start) == S.charAt(temp_end)) {
                      temp_start++;
                      temp_end--;
                      continue;
                  }
                  end--;
                  start--;
                  break;
              }

            }
        }
        return 1;
    }

    private static boolean isPal(String S) {
        for (int i=0; i<S.length(); i++) {
            if (S.charAt(i) != S.charAt(S.length()-i-1)) {
                return false;
            }
        }
        return true;
    }
}
